### PR TITLE
feat(bazel-module): Add support of maven methods

### LINKED ISSUE #27601: Bazel: support use_extension for maven
### Describe the proposed change(s).

Requested in https://github.com/renovatebot/renovate/discussions/27467

Reproduction can be found in https://github.com/renovate-reproductions/27467/blob/main/MODULE.bazel

### PR DESCRIPTION
**Pull request by al-babych-fivetran**
## Changes

[jvm_rules_external](https://github.com/bazelbuild/rules_jvm_external/blob/master/docs/bzlmod.md) with Bzlmod support was added.

## Context

Closes #27601 

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

Repository for test: https://github.com/al-babych-fivetran/renovate-maven-bzlmod

### CODE REVIEW

File: lib/modules/manager/bazel-module/maven.ts [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+import {
+  RecordFragmentSchema,
+  StringArrayFragmentSchema,
+  StringFragmentSchema,
+} from './fragments';
+
+const artifactMethod = 'artifact';
+const installMethod = 'install';
+const commonDepType = 'maven_install';
+export const mavenVariableRegex = /^maven.*/;
```
Comment (zharinov) [inline lib/modules/manager/bazel-module/maven.ts:14]: For this and other regex usages, please:

```suggestion
export const mavenVariableRegex = regEx(/^maven.*/);
```

File: lib/modules/manager/bazel-module/parser.ts [hunk 2]
```diff
@@ -11,29 +16,54 @@ const supportedRules = [
...
   );
 
 const moduleRules = q
-  .sym<Ctx>(supportedRulesRegex, (ctx, token) => ctx.startRule(token.value))
+  .opt<Ctx>(
+    q.sym(mavenVariableRegex, (ctx, token) => {
+      return ctx.startRule(token.value);
+    }),
+  )
+  .opt(q.op('.'))
```
Comment (zharinov) [inline lib/modules/manager/bazel-module/parser.ts:52]: Maven-specific matcher should be extracted into `mavenRules` and composed with the `moduleRules` via `alt()` (see `const rule = q.alt<Ctx>(moduleRlues);`).

File: lib/modules/manager/bazel-module/parser.ts [hunk 3]
```diff
@@ -51,3 +81,14 @@ export function parse(input: string): RecordFragment[] {
...
+function extendMavenRule(record: RecordFragment, tokenValue: string): void {
+  const rule = record.children.rule;
+  if (
+    rule.type === 'string' &&
+    rule.value.match(mavenVariableRegex) &&
+    bzlmodMavenMethods.includes(tokenValue)
+  ) {
+    rule.value = getParsedRuleByMethod(tokenValue);
+  }
+}
```
Comment (zharinov) [inline lib/modules/manager/bazel-module/parser.ts:94]: Maybe just put it to the `mavenRules`, once it's separated from `moduleRules`

**Review by zharinov — CHANGES_REQUESTED** [review]
Maven-related parsing should live in the separate file, see #30945

**Review by zharinov — APPROVED** [review]

**Review by rarkins — APPROVED** [review]

**Pull request merged**